src/main/resources/static/javascript/mainApp/general.controller.js

```diff
@@ -262,6 +265,19 @@
         $scope.hoverCopy = () => {
             $("[data-content='copy']").popover();
         };
+
+        /**
+         * Display a modal when an invalid email is entered on the feedback page
+         */
+        $scope.checkEmail = (event) => {
+            if (!($scope.emailPattern.test($scope.feedbackEmail))) {
+                if(event) {
+                    $scope.displayDynamicModal(Message.Title.INVALID_EMAIL_ERROR, Message.Body.INVALID_EMAIL_ERROR);
+                    event.preventDefault();
+                    return;
+                }
+            }
+        };
```

Comment:
It looks like you can combine the two if statements with `&&`. Also the line with `return;` might not be needed, not completely sure.